testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -1048,26 +1067,40 @@ public void testOutputColumnsForSetOperations(String setOperator)
                 {"EXCEPT ALL"}};
     }
 
-    private void assertColumnLineage(String baseQuery, OutputColumnMetadata... outputColumnMetadata)
+    private void assertLineage(String baseQuery, Set<String> inputTables, OutputColumnMetadata... outputColumnMetadata)
             throws Exception
     {
         runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table AS " + baseQuery, 2);
-        assertColumnMetadata(outputColumnMetadata);
+        assertLineage(inputTables, outputColumnMetadata);
 
         runQueryAndWaitForEvents("CREATE VIEW mock.default.create_new_view AS " + baseQuery, 2);
-        assertColumnMetadata(outputColumnMetadata);
+        assertLineage(inputTables, outputColumnMetadata);
 
         runQueryAndWaitForEvents("CREATE VIEW mock.default.create_new_materialized_view AS " + baseQuery, 2);
-        assertColumnMetadata(outputColumnMetadata);
+        assertLineage(inputTables, outputColumnMetadata);
 
         runQueryAndWaitForEvents("INSERT INTO mock.default.table_for_output(test_varchar, test_bigint) " + baseQuery, 2);
```

Comment:
Please also add tests for UPDATE and DELETE (for completeness)

Reply:
Applying this change for `UPDATE` and `DELETE` is a bit tricky and can we address them in a different PR ?

Reply:
Sure.